qiita_db/analysis.py

```diff
@@ -753,7 +753,7 @@ def _add_file(self, filename, filetype, data_type=None, conn_handler=None):
                "checksum, checksum_algorithm_id, data_directory_id) VALUES "
                "(%s, %s, %s, %s, %s) RETURNING filepath_id")
         # magic number 1 is for crc32 checksum algorithm
-        fpid = conn_handler.execute_fetchone(sql, (fullpath, fptypeid,
+        fpid = conn_handler.execute_fetchone(sql, (filename, fptypeid,
```

Comment:
This line looks ok. However, I'm looking at the function in general `_add_file` and it looks pretty similar to `insert_filepaths` in the `util.py` file. Is it possible to remove this one and use the `insert_filepaths` function here, so no code duplication happens in any place? BTW, I would say that the code duplication was the source of this bug...